### PR TITLE
remote_cache: Fix redis

### PR DESCRIPTION
**Pull request by kylebrandt**
Making draft because:

1. Want to know if I should be flipping the expire duration from negative to positive or if it being negative in the first place is mistake?
2. If instead of parsing the connection string, we should at DB specific options to the .ini for remote_cache?
3. Need to fix whatever check is failing

### CODE REVIEW

**Comment by kylebrandt** [comment]
Test failure is related to negative thing - will discuss with Carl on Monday.

File: pkg/infra/remotecache/redis_storage.go [hunk 1]
```diff
@@ -13,12 +16,47 @@ type redisStorage struct {
...
+		connVal := keyValueTuple[1]
+		switch connKey {
+		case "addr":
+			options.Addr = connVal
+		case "password":
+			options.Password = connVal
+		case "db":
+			i, err := strconv.ParseInt(connVal, 10, 64)
+			if err != nil {
+				return nil, fmt.Errorf("value for db in redis connection string must be a number: %v", err)
```
Comment (bergquist) [inline pkg/infra/remotecache/redis_storage.go]: https://github.com/grafana/grafana/blob/master/pkg/util/errutil/errors.go#L13

**Review by bergquist — APPROVED** [review]
LGTM

**Comment by bergquist** [comment]
This is breaking for current redis users. While its few.. we need to highlight that in the changelog.

**Pull request merged**